concourse/pipelines/create-cloudfoundry.yml

```diff
@@ -2127,6 +2221,67 @@ jobs:
                   cf set-env rubbernecker PIVOTAL_PROJECT_ID ${PIVOTAL_PROJECT_ID}
                   cf start rubbernecker
 
+        - task: deploy-paas-metrics
+          config:
+            platform: linux
+            image_resource:
+              type: docker-image
+              source:
+                repository: governmentpaas/cf-cli
+                tag: 895cf6752c8ec64af05a3a735186b90acd3db65a
+            params:
+              DISABLE_METRICS: ${DISABLE_METRICS}
+              DATADOG_API_KEY: {{datadog_api_key}}
+              DATADOG_APP_KEY: {{datadog_app_key}}
+              SYSTEM_DNS_NAME: {{system_dns_zone_name}}
+              DEPLOY_ENV: {{deploy_env}}
+            inputs:
+              - name: paas-cf
+              - name: config
+              - name: cf-secrets
+              - name: cf-manifest
+              - name: bosh-CA
+            run:
+              path: sh
+              args:
+                - -e
+                - -u
+                - -c
+                - |
+                  VAL_FROM_YAML=$(pwd)/paas-cf/concourse/scripts/val_from_yaml.rb
+                  METRICS_USERNAME=paas-metrics
+                  METRICS_PASSWORD=$($VAL_FROM_YAML secrets.paas_metrics_password cf-secrets/cf-secrets.yml)
+                  ./paas-cf/concourse/scripts/import_bosh_ca.sh
+
+                  . ./config/config.sh
+                  echo | cf login -a ${API_ENDPOINT} -u ${CF_ADMIN} -p ${CF_PASS} -o admin -s monitoring
+
+                  BUILD_ROOT=$(pwd)
+                  cd paas-cf/tools/metrics
+
+                  ruby -ryaml -e "
+                    env = {
+                      'CF_USER' => '${METRICS_USERNAME}',
+                      'CF_PASS' => '${METRICS_PASSWORD}',
+                      'CF_API_ADDRESS' => '${API_ENDPOINT}',
+                      'CF_SKIP_SSL_VALIDATION' => 'true',
+                      'DATADOG_API_KEY' => '${DATADOG_API_KEY}',
+                      'DATADOG_APP_KEY' => '${DATADOG_APP_KEY}',
+                      'DEPLOY_ENV' => '${DEPLOY_ENV}',
+                    }
+                    manifest = YAML.load_file('manifest.yml')
+                    manifest['applications'].each { |app|
+                      app['env'] = {} unless app['env']
+                      app['env'].merge!(env)
+                    }
+                    File.write('manifest.yml', manifest.to_yaml)
+                  "
+
+                  if  [ "${DISABLE_METRICS:-}" != "true" ] ; then
```

Comment:
Could we move this if statement to one of the first lines of this task? We probably don't need all that prep work to be done when we're not going to end up deploying this.